### PR TITLE
Add wait for checking the availability of pods and services

### PR DESCRIPTION
**Pull request by STEPHINRACHEL**
**What**
Added wait function for checking the availability of services.

### CODE REVIEW

File: deploy/deploy.sh [hunk 1]
```diff
@@ -53,13 +51,13 @@ oc apply -f $DIR/amq/address.yaml -n $NS
 
 echo "Waiting for AMQ AddressSpace to be ready"
 # unfortunately oc wait doesn't work for addressspace and address types (problem with AMQ itself)
-retry 30 5 oc get addressspace/workload-app -n $NS -o 'jsonpath={.status.isReady}' | grep 'true'
+wait_for "oc get addressspace/workload-app -n $NS -o 'jsonpath={.status.isReady}' | grep -q 'true'" "Addressspace is ready" "5m" "60"
```
Comment (wei-lee) [inline deploy/deploy.sh]: this will sleep for 60 seconds between retries? That seems a bit too long I think?

File: deploy/deploy.sh [hunk 2]
```diff
@@ -80,6 +78,8 @@ oc create secret generic rhsso-secret --from-literal=RHSSO_PWD=$RHSSO_PWD --from
 
 # Deploy 3scale Resources
 THREE_SCALE_URL=$(${DIR}/3scale.sh deploy)
+echo "Waiting for the ${THREE_SCALE_URL} to be reachable"
+wait_for "curl -s -o /dev/null -I -w '%{http_code}' ${THREE_SCALE_URL} | grep  -q 200" "Wait until api available" "10m" "120"
```
Comment (wei-lee) [inline deploy/deploy.sh]: If we are waiting for 3scale to be ready here, then I don't think we need to wait for another 2 minutes at the end: L109. we can probably remove that.

File: deploy/deploy.sh (hunk 1 above)
Comment (STEPHINRACHEL) [reply deploy/deploy.sh]: Is it fine to make it 30s?

Comment (wei-lee) [reply deploy/deploy.sh]: probably 10 seconds should be enough, it shouldn't take that long for them to get ready. We should update them in the rest of the file too

**Pull request merged**